cocotb/scheduler.py

```diff
@@ -585,14 +585,19 @@ def wrapper():
             except BaseException as e:
                 _outcome = outcomes.Error(e)
             event.outcome = _outcome
+            # We need to make sure that thread_resume has been called for the
+            # calling thread so that the scheduler will return the outcome to
+            # the calling thread before advancing the sim
```

Comment:
I think this would be clearer with a remark like:

> Ensure we notify the current (scheduler) thread that we are about to wake up the background thread,
> making sure to do so before the background thread gets a chance to notify that it is waking again
> via thread_suspend

I'll let you decide the right compromise between your comment and that suggestion

Reply:
I merged our comments and force-pushed.